### PR TITLE
Fix outgoing lw for history and coupling.

### PR DESCRIPTION
**Pull request by dabail10**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [X] Short (1 sentence) summary of your PR: 
A bug was identified when a grid cell goes from ice free to ice covered. Issue #994
- [X] Developer(s): 
dabail10 (D. Bailey)
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#6aa677ec49b415298b6368a19e5122f9c9d49f8e
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR update the Icepack submodule?  If so, the Icepack submodule must point to a hash on Icepack's main branch.
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [X] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [X] No 
- [X] Please document the changes in detail, including _why_ the changes are made.  This will become part of the PR commit log.
This is a two-line change in ice_flux.F90 (scale_fluxes) to check for the condition where aice > ), but flwout > -puny. This means the cell was ice free at the beginning of the step, but non-zero at the end. It sets the outgoing longwave to the open ocean value based on the freezing temperature. This is bfb, but will change the history output of flwup and flwup_ai. Test results are coming.

### CODE REVIEW

**Comment by dabail10** [comment]
Here are the test results:

https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#6aa677ec49b415298b6368a19e5122f9c9d49f8e

File: cicecore/cicedyn/general/ice_flux.F90 [hunk 1]
```diff
@@ -1231,7 +1231,10 @@ subroutine scale_fluxes (nx_block, ny_block, &
             fsens   (i,j) = fsens   (i,j) * ar
             flat    (i,j) = flat    (i,j) * ar
             fswabs  (i,j) = fswabs  (i,j) * ar
+            if (flwout(i,j) > -puny) & 
+               flwout  (i,j) = -stefan_boltzmann *(Tf(i,j) + Tffresh)**4
             flwout  (i,j) = flwout  (i,j) * ar
+            ! Special case where aice_init was zero and aice > 0.
```
Comment (eclare108213) [inline cicecore/cicedyn/general/ice_flux.F90:1237]: why is the comment 2 lines below the if statement?

Comment (dabail10) [reply cicecore/cicedyn/general/ice_flux.F90]: Whoops. Fixed now.

**Comment by anton-seaice** [comment]
If its ok, ill defer to Tony and Elizabeth on this. ACCESS don't use this field.

**Comment by dabail10** [comment]
> If its ok, ill defer to Tony and Elizabeth on this. ACCESS don't use this field.

Really? Was does the atmosphere component use for the upward longwave computation?

**Comment by dabail10** [comment]
> Its calculated internally in the atmosphere - presumably in the same way CICE is calculating it

This probably is fine as long as the surface temperature is correct for the sea ice.

**Comment by dabail10** [comment]
The shortwave fluxes and albedo are computed at the end of the step with the updated ice fraction. This is to be synchronized with the shortwave coming from the atmosphere on the next step. However, we could potentially revisit this. I'm doing it in scale_fluxes because it is where flwout is either divided by aice or it is set to the freezing point longwave. I agree there is a larger issue an I am happy to open one for this.

**Comment by eclare108213** [comment]
flwout is initialized in suboutine init_coupler_fluxes [here](https://github.com/CICE-Consortium/CICE/blob/749e0742c0323cafe1723974b9e5ea4a6cc31bbe/cicecore/cicedyn/general/ice_flux.F90#L767). This is currently the freshwater value, intended over ice. Would it work to alternatively initialize it here for seawater Tf in an if block based on aice, rather than in scale_fluxes?

**Comment by dabail10** [comment]
This is the whole issue with the bug. It is indeed initialized in init_coupler_fluxes, but then it is overwritten in init_flux_atm to be zero!

**Comment by eclare108213** [comment]
Then why not initialize it in init_flux_atm?  Sorry I'm missing something here, maybe in the order that things are done in the various drivers.

**Comment by dabail10** [comment]
It's an interesting question. The reason things are initialized in init_flux_atm as they are category merged quantities. So, flwout = flwout + sum(aicen(n)*flwoutn(n)). True also of fsens, flat, etc. However, if there is no ice at the beginning of the timestep then it is never computed and hence the issue with the bug. I tried this originally with something like:

where (aice(:,:,:) > 0) then
    flwout(:,:,:) = 0
endwhere

Then flwout would only be zeroed out when there was ice at the beginning of the step. This did not quite work as expected though. I don't understand why this was done in both init_flux_atm and init_flux_coupler. My main goal is to fix an actual bug where flwout = c0 when aice > 0. I am open to adding something more general later on.

**Review by eclare108213 — APPROVED** [review]
I'm approving this as a bug fix, but please create an issue referencing this PR, to thoroughly examine and document the coupling fields and steps.

**Review by apcraig — APPROVED** [review]

**Pull request merged**